### PR TITLE
fix: create a bazel-out node_modules tree using copy_file in the external repo when exports_directories_only is True

### PR DESCRIPTION
**Pull request by gregmagolan**
This will allow entry points to resolve peer node_module deps within an external yarn_install & npm_install repository

### CODE REVIEW

File: internal/npm_install/generate_build_file.ts [hunk 1]
```diff
@@ -51,6 +51,8 @@ const PUBLIC_VISIBILITY = '//visibility:public';
 
 let LEGACY_NODE_MODULES_PACKAGE_NAME = '$node_modules$';
 
+const WORKSPACE_REROOTED_PATH = '_';
```
Comment (alexeagle) [inline internal/npm_install/generate_build_file.ts:54]: strangely descriptive name for this constant that's one character, and unused?

File: internal/npm_install/generate_build_file.ts [hunk 2]
```diff
@@ -206,7 +214,7 @@ function flattenDependencies(pkgs: Dep[]) {
 async function generateRootBuildFile(pkgs: Dep[]) {
   let buildFile = config.exports_directories_only ?
       printRootExportsDirectories(pkgs) :
-      printRoot(pkgs);
+      printRootLegacy(pkgs);
```
Comment (alexeagle) [inline internal/npm_install/generate_build_file.ts:217]: maybe "ExportsAllFiles" is more descriptive than "Legacy"?

**Review by alexeagle — APPROVED** [review]
neat

File: internal/npm_install/generate_build_file.ts (hunk 1 above)
Comment (gregmagolan) [reply internal/npm_install/generate_build_file.ts]: was used at one point, does that count?

File: internal/npm_install/generate_build_file.ts (hunk 2 above)
Comment (gregmagolan) [reply internal/npm_install/generate_build_file.ts]: either or

**Pull request merged**